### ISSUE TITLE
Problem with HTML comments comparison 

### ISSUE THREAD

**Issue by rszachtsznajder** [rtfpessoa/diff2html#276]
I've tried to use latest diff2html (at this moment 3.0.0-master.aa6dd18). I found an issue related to HTML comparison. It doesn't work properly with HTML comments. Take a look on the example below.

**diff file:**

```
--- src/index.html
+++ src/index.html
@@ -1,2 +1,2 @@
-<!-- commented code -->
-</div>
+<html>
+<body>
```

**result is printed as:**

![diff2html-issue](https://user-images.githubusercontent.com/18330466/71851462-ed43ce80-30d6-11ea-9c4e-c4fa6ec83dc6.png)

**configuration**

```
{ 
    drawFileList: false, 
    matching: LineMatchingType.LINES, 
    diffStyle: DiffStyleType.WORD,
}
```

After some time of debugging I found that the issue concerns also other HTML elements that starts with "<" followed by non-letter character (for example `<!DOCTYPE html>` - here is "!" sign after "<").

**Comment by rtfpessoa** [comment]
@rszachtsznajder that seems like a nasty bug. Not sure how it passed through the tests.
Thanks for reporting, I will try to see what is the issue and get a fix soon.
If you have any time PRs are always welcome.

**Comment by rtfpessoa** [comment]
I think I got the reason. Seems like my refactor to achieve more standardized escaping caused some issues with the diff difference highlight between lines.

```html
<!-- Correct -->
<span class="d2h-code-line-ctn"><del>&lt;!-- commented code --&gt;</del></span>
<!-- Wrong -->
<span class="d2h-code-line-ctn">&lt;<del>;!-- commented code --&amp;</del>gt;</span>
```

**Comment by rtfpessoa** [comment]
Released fix as version `3.0.0-master.746d3e6`

**Comment by rtfpessoa** [comment]
Thanks a lot for reporting ! :+1:
